widgetsnbextension/src/manager.js

```diff
@@ -49,6 +49,18 @@ function new_comm(manager, target_name, data, callbacks, metadata, comm_id, buff
     return manager.new_comm.apply(manager, Array.prototype.slice.call(arguments, 1));
 }
 
+function filter_existing_model_state(manager, state) {
+    var models = state.state;
+    models = Object.keys(models)
+          .filter(function(model_id) {
+              return !manager._models[model_id];
+          })
+          .reduce(function(res, model_id) {
+              return Object.assign(res, { [model_id]: models[model_id] });
+          }, {});
+    return Object.assign({}, state, {state: models});
```

Comment:
Can you use Backbone's _.extend? Object.assign is not supported in IE, and we support IE in the classic notebook, IIRC.